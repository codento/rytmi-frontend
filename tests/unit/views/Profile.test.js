jest.mock('lodash', () => ({
  orderBy: jest.fn(() => {
    return [
      {
        name: 'Vue.js',
        knows: 0,
        wantsTo: 2,
        id: 7
      }
    ]
  })
}))

/* eslint-disable */
import Vuex from 'vuex'
import BootstrapVue from 'bootstrap-vue'
import { shallowMount, createLocalVue } from '@vue/test-utils'
import View from '@/views/Profile.vue'
import lodash from 'lodash'
import Loading from '@/components/helpers/LoadingSpinner'
/* eslint-enable */

describe('Profile.vue shallowMount', () => {
  let localVue = createLocalVue()
  localVue.use(Vuex)
  localVue.use(BootstrapVue)
  localVue.component('loading', Loading)

  let store, getters, cmp
  let $lodash = lodash

  const $route = {
    path: '/id',
    params: { id: 5 }
  }
  beforeEach(() => {
    getters = {
      profileId: () => (arg) => arg,
      profileById: () => (arg) => {
        return {
          id: arg,
          userId: 1,
          firstName: 'foo',
          lastName: 'bar',
          photoPath: '',
          title: 'software developer',
          accounts: [
            {
              address: 'twitter.com/foo'
            },
            {
              address: 'github.com/bar'
            }
          ],
          email: 'foo.bar@foo.com',
          phone: '1354',
          introduction: { fi: 'intro in Finnish', en: 'intro in English' }
        }
      },
      skillById: () => (skillId) => {
        return {
          id: skillId,
          name: 'Python',
          description: 'Python desc'
        }
      },
      profileSkillsByProfileId: () => (profileId) => {
        return [
          {
            id: 1,
            name: 'Python',
            description: 'Python desc'
          }
        ]
      },
      profiles: () => (arg) => arg,
      skillsById: () => (arg) => arg,
      profileProjectsByProfileId: () => (profileId) => {
        return [
          {
            id: 1,
            profile: profileId,
            projectId: 1,
            startDate: '2018-01-01',
            endDate: '2018-02-01'
          }
        ]
      },
      projectById: () => (projectId) => {
        return {
          id: projectId,
          name: { en: 'Project Foo (en)', fi: 'Projekti Foo' },
          customerName: { en: 'Customer', fi: 'Asiakas' },
          description: { en: 'Foo Bar (en)', fi: 'Foo Bar' }
        }
      },
      profileEmployersByProfileId: () => (profileId) => {
        return [
          {
            profileId,
            employerId: 1
          }
        ]
      },
      employers: () => () => {
        return [
          {
            name: 'Codento Oy',
            id: 1
          }
        ]
      }
    }
    store = new Vuex.Store({ getters })
    cmp = shallowMount(View, {
      store,
      localVue,
      mocks: {
        $lodash,
        $route
      }
    })
  })
  it('Template is correct', () => {
    expect(cmp.element).toMatchSnapshot()
  })
})
