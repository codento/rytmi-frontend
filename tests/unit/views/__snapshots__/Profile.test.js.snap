// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile.vue shallowMount Template is correct 1`] = `
<b-container-stub
  class="animated fadeIn profile mt-2"
  tag="div"
>
  <b-row-stub
    tag="div"
  >
    <b-col-stub
      tag="div"
    >
      <b-btn-stub
        class="open-cv-tool-button pull-right"
        event="click"
        routertag="a"
        tag="button"
        target="_self"
        type="button"
        variant="secondary"
      >
        
        Open CV tool
      
      </b-btn-stub>
    </b-col-stub>
     
    <userprofile-stub
      profile="[object Object]"
    />
     
    <cvtool-stub
      profile="[object Object]"
    />
  </b-row-stub>
</b-container-stub>
`;
